src/components/shared/editable-text-input-list/editable-text-input-list.tsx

```diff
@@ -18,21 +20,69 @@ const EditableTextInputList: React.FC<EditableTextInputListProps> = ({
   remove,
   update
 }) => {
+  const [lastInput, setLastInput] = useState(
+    items.length > 0 ? items[items.length - 1].toString() : ''
+  );
+  const focusRef = useRef<HTMLInputElement>(null);
+
+  /**
+   * Handles key presses in the form control.
+   * @param e the event of the key press
+   * @param index the index of the input being pressed within the items list
+   */
+  const handleKeyDown = (e: any, index: number) => {
+    if (e.key === 'Enter') {
+      if (lastInput) {
+        addButtonOnClick();
+      }
+      focusRef.current?.focus();
+    }
+  };
+
+  /**
+   * On click function for the add button.
+   */
+  const addButtonOnClick = () => {
+    add('');
+    setLastInput('');
+  };
+
+  /**
+   * Checks if the given index is the last element of items.
+   * @param index the given index
+   * @returns true if the index is the last element of items
+   */
+  const isLastElement = (index: number) => {
+    return index === items.length - 1;
+  };
```

Comment:
I'm running into a bug where if you have all the items in `EditableTextInputList` except for the last (bottom) element filled in, delete the last element, focus on the new last element, and try to create a new entry by hitting 'Enter', I can't make a new entry. I also tried hitting 'Enter' from the top item in this same set up and that brings me to the lowest item, but then I run into the same aforementioned issue. I think the functionality here should create a new item regardless if the items are filled in or not, though we should probably get PMs or @jamescd18 to confirm this behavior. I think it has to do with the logic used to determine what the last element is, though I'm not entirely sure what the fix would be yet. Let me know if you need help recreating the bug I'm getting, otherwise everything else lgtm.

Reply:
Just added a commit that fixes this, it had to do with not updating the lastInput state to be the now second to last one when removing the last input box.